src/gui/static/src/app/services/exchange.service.ts

```diff
@@ -1,49 +1,158 @@
 import { throwError as observableThrowError, Observable, BehaviorSubject, SubscriptionLike, of } from 'rxjs';
+import { map, mergeMap, retryWhen, delay, catchError, tap } from 'rxjs/operators';
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
-import {
-  ExchangeOrder,
-  StoredExchangeOrder,
-  TradingPair,
-} from '../app.datatypes';
-import { StorageService, StorageType } from './storage.service';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
 import * as moment from 'moment';
-import { ApiService } from './api.service';
+
+import { StorageService, StorageType } from './storage.service';
 import { environment } from '../../environments/environment';
-import { map, mergeMap, retryWhen, delay, catchError, tap } from 'rxjs/operators';
 import { processServiceError } from '../utils/errors';
+import { OperationError } from '../utils/operation-error';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163600)
```suggestion
import { OperationError } from "../utils/operation-error";
```